### PR TITLE
json subtypes support

### LINKED ISSUE #112: @JsonTypeInfo and @JsonSubTypes are ignored and documentation is not generated for polymorphic types
We have polymorphic types as part of our API spec. There is an interface and there are abstract classes and concrete classes that implement the interface and auto-rest-docs appears to ignore the `@JsonTypeInfo` and `@JsonSubTypes` annotations. In the documentation that is generated, the field where the interface is used is correctly identified as an "Object". However no documentation is generated for the implementations. What are we missing? Any help is appreciated.

### PR DESCRIPTION
**Pull request by jmisur**
fixes #112

### CODE REVIEW

**Comment by jmisur** [comment]
added new endpoint to example project

**Comment by jmisur** [comment]
@fbenz pls check again

File: spring-auto-restdocs-core/src/main/java/capital/scalable/restdocs/snippet/StandardTableSnippet.java [hunk 1]
```diff
@@ -45,11 +45,8 @@ protected StandardTableSnippet(String snippetName, Map<String, Object> attribute
     @Override
     protected Map<String, Object> createModel(Operation operation) {
         HandlerMethod handlerMethod = getHandlerMethod(operation);
-
         Collection<FieldDescriptor> fieldDescriptors = emptyList();
-        if (handlerMethod != null) {
```
Comment (fbenz) [inline spring-auto-restdocs-core/src/main/java/capital/scalable/restdocs/snippet/StandardTableSnippet.java]: Are you sure? I remember that I added after getting a NPE here, but I don't remember what the scenario was.

**Comment by jmisur** [comment]
I will test scenario where handlerMethod is null - I think it's when you are calling incorrect url and want to document it. In that case I'll do some general error handling with proper message instead of NPE coming from somewhere.

**Pull request merged**